jme3-examples/src/main/java/jme3test/terrain/PBRTerrainTest.java

```diff
@@ -0,0 +1,348 @@
+    private float grassScale = 24;
+    private float marbleScale = 64;
+    private float roadScale = 64;
+    
+    private AmbientLight ambientLight;
+    private DirectionalLight directionalLight;
+    private boolean isNight = false;
+    private final float dayLightIntensity = 1.0f;
+    private final float nightLightIntensity = 0.03f;
+    
+    private BitmapText keybindingsText;
+
+    public static void main(String[] args) {
+        PBRTerrainTest app = new PBRTerrainTest();
+        AppSettings s = new AppSettings(true);
+        
+        s.put("GammaCorrection", true);        
+                
+        app.setSettings(s);
+        app.start();
+       
+    }
+    
+    private final ActionListener actionListener = new ActionListener() {
+
+        @Override
+        public void onAction(String name, boolean pressed, float tpf) {
+            if (name.equals("triPlanar") && !pressed) {
+                triPlanar = !triPlanar;
+                if (triPlanar) {
+                    matTerrain.setBoolean("useTriPlanarMapping", true);
+                    // planar textures don't use the mesh's texture coordinates but real world coordinates,
+                    // so we need to convert these texture coordinate scales into real world scales so it looks
+                    // the same when we switch to/from tr-planar mode (1024f is the alphamap size)
+                    matTerrain.setFloat("AlbedoMap_0_scale", 1f / (1024f / dirtScale));
+                    matTerrain.setFloat("AlbedoMap_1_scale", 1f / (1024f / darkRockScale));
+                    matTerrain.setFloat("AlbedoMap_2_scale", 1f / (1024f / snowScale));
+                    matTerrain.setFloat("AlbedoMap_3_scale", 1f / (1024f / riverRockScale));
+                    matTerrain.setFloat("AlbedoMap_4_scale", 1f / (1024f / grassScale));
+                    matTerrain.setFloat("AlbedoMap_5_scale", 1f / (1024f / marbleScale));
+                    matTerrain.setFloat("AlbedoMap_6_scale", 1f / (1024f / roadScale));
+                } else {
+                    matTerrain.setBoolean("useTriPlanarMapping", false);
+
+                    matTerrain.setFloat("AlbedoMap_0_scale", dirtScale);
+                    matTerrain.setFloat("AlbedoMap_1_scale", darkRockScale);
+                    matTerrain.setFloat("AlbedoMap_2_scale", snowScale);
+                    matTerrain.setFloat("AlbedoMap_3_scale", riverRockScale);
+                    matTerrain.setFloat("AlbedoMap_4_scale", grassScale);
+                    matTerrain.setFloat("AlbedoMap_5_scale", marbleScale);
+                    matTerrain.setFloat("AlbedoMap_6_scale", roadScale);
+
+                }
+            }
+            if (name.equals("toggleNight") && !pressed) {
+                isNight = !isNight;
+                //ambient and direcitonal light are faded smoothly in update loop below !
+                
+            }
+        }
+    };
+    
+    @Override
+    public void simpleInitApp() {
```

Comment:
`simpleInitApp()` is too long for easy comprehension. Please split it into smaller methods.

Reply:
> Both test apps need class javadoc to explain what they test and how they differ from one another.

Do you have a reference I could use to base this off of? I have never generated a javadoc for a testcase, so I'm unsure exactly how this should look.

And my apologies for leaving so much unorganized/unoptimized code. I was unaware that official test cases needed to be. I believe these test cases were based off of this one which is also not as great as it could be, but its also much older: https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-examples/src/main/java/jme3test/terrain/TerrainTestAdvanced.java

So I apologize for making this such a long review process for you, and I appreciate your patience. When I address your most recent reviews, I will also take a slightly more careful look through the test cases to make sure the code is easy to understand and organized.

Reply:
+ Code is never perfect, and that's not the standard being applied here.
+ Since these tests are not automated, they are useless if the person running them doesn't know what they are supposed to do and what results/behaviors to expect. 
+ Test cases have to be maintained over time. Since I doubt you'll be the one maintaining them, it's important that they should be easy to understand and not brittle.
+ These test cases are also examples (hence the name jme3-examples) which will help developers figure out how the PBR terrain materials are supposed to be used. Official examples should ... set a good example.

For a good example of class javadoc in jme3-examples, see https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-examples/src/main/java/jme3test/bullet/shape/TestGimpactShape.java

Reply:
Yes I don't think I was right to use the word perfect, and I don't want it to sound like I am contesting your requests to make a good test case. I just was using that other test case mine were based off of as the standard, and in hindsight I should have just put in the effort from the get go to make the test cases better from the start to save you the time of having to review everything so much.

Reply:
The JME repo contains plenty of crappy code. If such code shaped your expectations, I can see why you might be surprised by some of my feedback.

Perhaps a more experienced reviewer than myself would've caught more issues on each review pass. Unfortunately, the project's more experienced reviewers seem to have other priorities these days, so I've had to fill their shoes. I know I've learned much from this process, and I hope you're learning also.

We're close to integrating this PR. I'm looking forward to it!